packages/hotpotqa/src/aviary/envs/hotpotqa/env.py

```diff
@@ -340,6 +343,8 @@ def finish(self, answer: str) -> str:
 
         self.state.answer = answer
         self.state.reward += self.calculate_reward(answer)
+
+        self.state.last_action = self.tools[2].info.name
```

Comment:
I saw @jamesbraza 's original suggestion, but IMO this is less readable and depends on the order in which `self.tools` is populated. "Finish" is better. If we want to be extra careful, we could make a constant `FINISH_TOOL_NAME = "Finish"` and use it in both places.

Reply:
Yeah I agree with Sid that we shouldn't rely on ordering here @albertbou92 . Can you find a way to:
- Not depend on tool ordering
- Not depend on string literals (as subclasses can change tool names)

Reply:
Maybe just `last_action_is_lookup: bool`, since that's the main thing we check with `last_action`?